docs/blog/20240624-Daeraxa-3kStars.md

```diff
@@ -0,0 +1,136 @@
+We had some big tasks ahead of us; changes needed to be made, and work needed to start to get some traction on the project. We desperately wanted to keep the community alive, but with the announcement of the sunset, people were fleeing the sinking ship. We had to get the project up and running to show that Pulsar was a serious continuation of the project and not just a fork on GitHub with a renamed readme.
+
+- **Building from source** - Atom had a rather complicated (and now rather fragile) way of [bootstrapping](https://atom-flight-manual-archive.github.io/hacking-atom/sections/hacking-on-atom-core/#cloning-and-bootstrapping) itself for development. One of the first tasks was to simplify this into a process much more familiar to those using Node.js and Electron, which is far more robust and makes it much easier to get involved with Pulsar development.
+- **Rebranding** - We had to start removing Atom references and replacing them with Pulsar ones. This was not an easy "find and replace" job as `atom` is used extensively within the APIs and code and therefore something we had to be careful of. We also had to decide on a new logo for the project and add it to the application, along with updates to redirect links within Pulsar to our new repositories, websites, forums, etc.
+- **Decaffeination** - Atom was originally written mostly in [CoffeeScript](https://coffeescript.org/), a rather useful language at the time that compiles to JavaScript. With more and more functionality being introduced into JavaScript, CoffeeScript was quickly becoming obsolete and actually started creating a barrier to entry for contribution rather than easing it, as was the initial intention. The original Atom team had started this process, but over time we were able to achieve the goal of complete decaffeination in Pulsar 1.109 and in PPM in 1.110.
+- **CI and build** - We needed an automated platform to build our binaries and run tests. For this, we went with [CirrusCI](https://cirrus-ci.org/) which had a more than generous free tier, and it reliably built our binaries and ran our tests for over a year. We have since had to [move away](https://pulsar-edit.dev/blog/20230903-confused-Techie-pulsars-ci.html) from CirrusCI.
+- **Versioning** - Atom's last (at the time) release was Atom 1.60, which left us with some issues. We wanted to stay with [Semantic Versioning](https://semver.org/) as Atom had, but what should our first full release be? We could reset the counter back to 1.0, but that would cause packages looking for a minimum version of Atom to break. We could move to 2.0, but again, there was an issue with packages, which were specifically looking only for 1.x versions, but this time as a *maximum* version. We could continue the numbering from Atom, but this seemed like we weren't getting a clean break as a new project. In the end, we decided to jump from 1.60 to 1.100 - a number that would give us a clean start whilst avoiding the issues with package version limits.
```

Comment:
```suggestion
- **Versioning:** - Atom's last (at the time) release was Atom 1.60. We wanted to stay with [Semantic Versioning](https://semver.org/) as Atom had — but what should our first full release be?

  It’s common for Atom packages to specify `atom` in their `package.json`’s `engines` field — and to specify a version range of `>=1.0 <2.0`. We could reset the counter back to 1.0 — but many packages specified a newer version of Atom than 1.0 because they relied on features that were added in a certain Atom version. We could move to 2.0 — but we would’ve broken lots of packages that had specified a *maximum* compatible version. We could continue the numbering from Atom — but this seemed like we weren't getting a clean break as a new project.
    
  In the end, we decided to jump from 1.60 to 1.100 — a number that would give us a clean start whilst avoiding the issues with package version limits.
    
```